src/asynciterable/asynciterablex.ts

```diff
@@ -1,24 +1,40 @@
 import { as as asAsyncIterable } from './as';
 import { _initialize as _initializeFrom } from './from';
 import { OperatorAsyncFunction, UnaryFunction } from '../interfaces';
-import { bindCallback } from '../util/bindcallback';
 import { Observable } from '../observer';
 import { isReadableNodeStream, isWritableNodeStream } from '../util/isiterable';
 
+class WithAbortAsyncIterable<TSource> implements AsyncIterable<TSource> {
+  private _source: AsyncIterable<TSource>;
+  private _signal: AbortSignal;
+
+  constructor(source: AsyncIterable<TSource>, signal: AbortSignal) {
+    this._source = source;
+    this._signal = signal;
+  }
+
+  [Symbol.asyncIterator](): AsyncIterator<TSource> {
+    // @ts-ignore
+    return this._source[Symbol.asyncIterator](this._signal);
```

Comment:
So this only works with `ix`'s AsyncIterable's then, right? I had hoped that `wrapWithAbort()` would work with any `AsyncIterable`, checking the signal before and after every `next()` call, and calling `this._source.error?.(new AbortError())` on cancellation (for async generators, which always have `error()` implemented and can be aborted through that)

Reply:
We could definitely have that as an adapter method. Passing the abort signal through the subscription chain is the best way to ensure cancellation is propagated everywhere, including things like internal Promises.